bundler/spec/runtime/platform_spec.rb

```diff
@@ -281,38 +281,67 @@
 
     expect(the_bundle).to include_gems "platform_specific 1.0 RUBY"
 
-    build_repo4 do
-      build_gem "libv8"
+    simulate_platform "x86_64-linux" do
+      build_repo4 do
+        build_gem "libv8"
 
-      build_gem "libv8" do |s|
-        s.platform = Bundler.local_platform
+        build_gem "libv8" do |s|
+          s.platform = "x86_64-linux"
+        end
       end
+
+      gemfile <<-G
+        source "#{file_uri_for(gem_repo4)}"
+        gem "libv8"
+      G
+
+      lockfile <<-L
+        GEM
+          remote: #{file_uri_for(gem_repo4)}/
+          specs:
+            libv8 (1.0)
+
+        PLATFORMS
+          ruby
+
+        DEPENDENCIES
+          libv8
+
+        BUNDLED WITH
+           #{Bundler::VERSION}
+      L
+
+      bundle "install"
+
+      expect(the_bundle).to include_gems "libv8 1.0 x86_64-linux"
     end
+  end
 
+  it "doesn't pull platform specific gems on truffleruby, even if lockfile only includes those", :truffleruby_only do
     gemfile <<-G
-      source "#{file_uri_for(gem_repo4)}"
-      gem "libv8"
+      source "#{file_uri_for(gem_repo1)}"
+      gem "platform_specific"
     G
 
     lockfile <<-L
       GEM
-        remote: #{file_uri_for(gem_repo4)}/
+        remote: #{file_uri_for(gem_repo1)}/
         specs:
-          libv8 (1.0)
+          platform_specific (1.0-x86-darwin-100)
```

Comment:
Do we need a simulate_platform around or it isn't necessary here?

Reply:
This is precisely to test the case where we have a lockfile generated on a different platform. Since we never test on `x86-darwin-100` (I don't think that exists at all), I think it works as is. If we wrapped things with `simulate_platform` the spec would change its meaning I believe?

Reply:
Well we could use `simulate_platform` just to run `bundle install` to generate the lockfile. But to make specs simpler, focus and what we want to test and don't run a lot of extra bundler process I usually favor writing whatever lockfile we're having problems with directly.

Reply:
OK, makes sense